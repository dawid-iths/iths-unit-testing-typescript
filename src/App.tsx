import './App.css';
import NavMenu from './components/NavBar/NavMenu';
import { Route,Routes } from 'react-router-dom';
import Home from './pages/Home';
import About from './pages/About';
import Counter from './pages/Counter';
import SnapShot from './pages/SnapShot';

function App() {
  return (
    <div className="App">
      <NavMenu />
      <header className="App-header">
        <Routes>
          <Route path="/" element={<Home />} />
          <Route path="/counter" element={<Counter />} />
          <Route path="/snapshot" element={<SnapShot />} />
        </Routes>
      </header>
    </div>
  );
}

export default App;
