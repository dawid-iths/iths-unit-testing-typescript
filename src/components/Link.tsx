import {useState} from 'react';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

interface LinkProps {
    page: string;
    children: React.ReactNode;
}

const Link : React.FC<LinkProps> = (props) => {
  const [status, setStatus] = useState(STATUS.NORMAL);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.NORMAL);
  };

  return (
    <a
      className={status}
      href={props.page || '#'}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {props.children}
    </a>
  );
}
export default Link;