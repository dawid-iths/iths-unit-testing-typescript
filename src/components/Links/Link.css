.DefaultLink {
    color: rgb(132, 228, 84);
}
.HoveredLink {
    color: rgb(255, 187, 1);
}