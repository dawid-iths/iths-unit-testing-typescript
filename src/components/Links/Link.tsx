import { useState } from 'react';
import "./Link.css";

enum STATUS {
  HOVERED = 'HoveredLink',
  DEFAULT = 'DefaultLink',
};

export interface LinkProps {
  page: string;
  children: React.ReactNode;
}


const Link: React.FC<LinkProps> = (props) => {
  const [status, setStatus] = useState(STATUS.DEFAULT);

  const onMouseEnter = () => {
    setStatus(STATUS.HOVERED);
  };

  const onMouseLeave = () => {
    setStatus(STATUS.DEFAULT);
  };

  return (
    <a
      className={status}
      href={props.page || '#'}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
    >
      {props.children}
    </a>
  );
}
export default Link;