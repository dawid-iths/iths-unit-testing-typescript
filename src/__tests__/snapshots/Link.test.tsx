import App from '../../App';
import renderer from 'react-test-renderer'
import Link from '../../components/Links/Link';

it('<link /> the class when hovered', () => {
  const component : any = renderer.create(
    <Link page="http://www.ithsdistans.se">iths distans</Link>,
  );
  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseEnter();
  });

  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
