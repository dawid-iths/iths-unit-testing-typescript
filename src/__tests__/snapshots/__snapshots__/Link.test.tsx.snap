// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<link /> the class when hovered 1`] = `
<a
  className="DefaultLink"
  href="http://www.ithsdistans.se"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  iths distans
</a>
`;

exports[`<link /> the class when hovered 2`] = `
<a
  className="HoveredLink"
  href="http://www.ithsdistans.se"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  iths distans
</a>
`;

exports[`<link /> the class when hovered 3`] = `
<a
  className="DefaultLink"
  href="http://www.ithsdistans.se"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  iths distans
</a>
`;
